src/Microsoft.Health.Dicom.Core/Features/Workitem/WorkitemDatasetValidator.Add.cs

```diff
@@ -3,96 +3,97 @@
 // Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
 // -------------------------------------------------------------------------------------------------
 
-using System;
-using System.Collections.Generic;
 using System.Globalization;
-using System.Linq;
-using EnsureThat;
 using FellowOakDicom;
 using Microsoft.Health.Dicom.Core.Extensions;
-using Microsoft.Health.Dicom.Core.Features.Query;
 using Microsoft.Health.Dicom.Core.Features.Store;
+using Microsoft.Health.Dicom.Core.Models;
 
 namespace Microsoft.Health.Dicom.Core.Features.Workitem
 {
     /// <summary>
     /// Provides functionality to validate a <see cref="DicomDataset"/> to make sure it meets the minimum requirement when Adding.
+    /// <see href="https://dicom.nema.org/medical/dicom/current/output/html/part04.html#sect_5.4.2.1">Dicom 3.4.5.4.2.1</see>
     /// </summary>
     public class AddWorkitemDatasetValidator : WorkitemDatasetValidator
     {
-        protected override void OnValidate(DicomDataset dicomDataset, string workitemInstanceUid)
+        // 2/1 requirements are treated as 1/1 since the value should be set by this point.
+        // 3/* and -/- requirements are ignored.
+        protected override void OnValidate(DicomDataset dataset)
         {
-            EnsureArg.IsNotNull(dicomDataset, nameof(dicomDataset));
+            // TODO: return all validation exceptions together
 
-            ValidateWorkitemInstanceUid(dicomDataset, workitemInstanceUid);
+            dataset.ValidateRequirement(DicomTag.TransactionUID, RequirementCode.TwoTwo);
+            ValidateEmptyValue(dataset, DicomTag.TransactionUID);
 
-            ValidateRequiredTags(dicomDataset);
+            // SOP Common Module
+            // TODO: validate character set
```

Comment:
Still TODO?

Reply:
Yes, we haven't tackled this across the product.